### ISSUE TITLE
Automate ownCloud update

### ISSUE THREAD

**Issue by ypid** [debops/ansible-owncloud#28]
Not sure if this is a planned feature for the ownCloud packages which we use. We could automate this by a dpkg hook. Any ideas?

**Comment by drybjed** [comment]
On DebOps-managed hosts, `unattended-upgrades` is configured by `debops.apt` role. If ownCloud is installed using a `.deb` package, you should at least get e-mails informing you about possible upgrades.

One of the things i would like to do in the future is split `debops.apt` into smaller roles, then a role that manages the updates could be used to configure automatic upgrade of specific packages. Unless that could be done now via `conf.d/` style directory somewhere, not sure.

One thing I would be wary of is changes to the ownCloud configuration on a package update, any possibility that Ansible-managed configuration is overwritten this way?

**Comment by ypid** [comment]
> One thing I would be wary of is changes to the ownCloud configuration on a package update, any possibility that Ansible-managed configuration is overwritten this way?

The update writes the version into the `config.php` other DebOps specific ownCloud configs are not touched. I have `/var/www/owncloud/config` under version control :wink:

I think it would be possible to do the db upgrade in a file under `/etc/apt/apt.conf.d`. `etckeeper` also hooks into this mechanism. I might look into this.

**Comment by Gomez** [comment]
What will happen if a new package (8.1.5) appear here (?):

```
http://download.opensuse.org/repositories/isv:/ownCloud:/community:/8.1/Debian_8.0/
```

Admin will get a email with information, or will `unattended-upgrades` update and ownCloud stuck in maintenance because the db upgrade is missing.

**Comment by ypid** [comment]
`unattended-upgrades` is not configured to do this automatically yet. The problem is that as of ownCloud 9.0.2, the db upgrade still needs to happen manually. This is on my TODO list to be automated as said [above](https://github.com/debops/ansible-owncloud/issues/28#issue-119039614).

**Comment by ypid** [comment]
Implemented by: https://github.com/debops/ansible-owncloud/pull/44 in case anyone wants to test it before #44 is merged.

**Comment by ypid** [comment]
I did some more testing of this feature implemented by #44 and it is working great so far with my test instances and I will continue to test this. However, I will not enable this as role default until my confidence in this feature grows a bit more because the upstream package maintainers had trouble with this in the past and disabled it.

If I don’t find further problems, I intend to reenable it for v0.3.0.
